functions/src/platform/alexa/handler/handlers-builder.js

```diff
@@ -118,5 +178,41 @@ module.exports = (actions) => {
             });
         },
       };
+    })
+    // it would catcha any intents
+    // TODO: 1) actually we could use this handler to handle all intents
+    // and replace handlers above
+    // 2) we should implement fallback in case when we haven't matched any intent
+    .concat({
+      intent: 'any',
+
+      canHandle: () => true,
+
+      handle: (handlerInput) => {
+        debug('catch all the rest');
+
+        const res = findHandlersByInput(actions, handlerInput);
+        if (!res) {
+          debug(`we haven't found any valid handler`);
+          return Promise.resolve();
+        }
+
+        const {handlers, name} = res;
+
+        debug(`begin handle intent "${name}"`);
+        return fetchAttributes(handlerInput)
+          .then((persistentAttributes) => {
+            debug('got persistent attributes:', util.inspect(persistentAttributes, {depth: null}));
+            const app = new App(handlerInput, persistentAttributes);
+            const handler = fsm.selectHandler(app, handlers);
```

Comment:
our implementation of FMS is very simpler https://github.com/internetarchive/internet-archive-google-action/blob/master/functions/src/state/fsm.js. we store current state somewhere, have a function to update it and match current state to appropriate handler.

But it could be possible that someday we would require some more advanced architecture to deal with multiple state. For example `Hierarchically nested states` or other.

I'm still thinking that the way like we (and Dialog flow and Alexa) implement dialog application is far from the current way to implement AI - LSTM for example.